src/Microsoft.TemplateSearch.Common/BlobStorageTemplateInfo.cs

```diff
@@ -158,70 +110,160 @@ IReadOnlyList<ITemplateParameter> ITemplateInfo.Parameters
         [Obsolete("This property is deprecated")]
         bool ITemplateInfo.HasScriptRunningPostActions { get; set; }
 
+        public IReadOnlyDictionary<string, string> TagsCollection { get; private set; } = new Dictionary<string, string>();
+
         public static BlobStorageTemplateInfo FromJObject(JObject entry)
         {
-            return entry.ToObject<BlobStorageTemplateInfo>();
-        }
-    }
+            BlobStorageTemplateInfo info = entry.ToObject<BlobStorageTemplateInfo>();
 
-    internal class CacheTag : ICacheTag
-    {
-        [JsonProperty(PropertyName = "ChoicesAndDescriptions")]
-        private IReadOnlyDictionary<string, string> _choices = new Dictionary<string, string>();
+            //read parameters
+            bool readParameters = false;
+            List<ITemplateParameter> templateParameters = new List<ITemplateParameter>();
+            if (entry.TryGetValue(nameof(Parameters), StringComparison.OrdinalIgnoreCase, out JToken parametersArray))
+            {
+                if (parametersArray != null)
+                {
+                    foreach (JObject item in parametersArray)
+                    {
+                        templateParameters.Add(new BlobTemplateParameter(item));
+                    }
+                }
+                readParameters = true;
+            }
 
-        [JsonIgnore]
-        public string? DisplayName => throw new NotImplementedException();
+            //try read tags and parameters - for compatibility reason
+            if (entry.TryGetValue("tags", StringComparison.OrdinalIgnoreCase, out JToken tagsCollection) && tagsCollection is JObject tagsObject)
+            {
+                Dictionary<string, string> tags = new Dictionary<string, string>();
+                info.TagsCollection = tags;
+                foreach (JProperty item in tagsObject.Properties())
+                {
+                    JObject tagObj = (JObject)item.Value;
```

Comment:
This would throw if json isn't as expected, right? Maybe
```
JObject tagObj = item.Value as JObject;
```

Reply:
this is already refactored, will be better in next commit